### PR TITLE
Fix wrong key for getHttpServerTrustStorePassword

### LINKED ISSUE #3639: ServerConfiguration returns incorrect property for httpServerTrustStorePassword
**BUG REPORT**

***Describe the bug***

ServerConfiguration.getHttpServerTrustStorePassword() returns value of HTTP_SERVER_KEY_STORE_PASSWORD. Instead it should return the value of HTTP_SERVER_TRUST_STORE_PASSWORD.

***To Reproduce***

1. Set the following properties, specifying different key store and truststore files with different passwords:
httpServerTlsEnable=true
httpServerKeyStorePath
httpServerKeyStorePassword
httpServerTrustStorePath
httpServerTrustStorePassword
2. Start bookkeeper and observe the truststore password is incorrect.

***Expected behavior***

The truststore should be unlocked with its password.

***Screenshots***

N/A

***Additional context***

N/A

### PR DESCRIPTION
**Pull request by wenbingshen**
### Motivation
Fix #3639 

As the title.

### CODE REVIEW

**Review by Shoothzj — APPROVED** [review]

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java [hunk 1]
```diff
@@ -3663,7 +3663,7 @@ public ServerConfiguration setHttpServerTrustStorePath(String trustStorePath) {
      * @return http server trustStore password
      */
     public String getHttpServerTrustStorePassword() {
-        return getString(HTTP_SERVER_KEY_STORE_PASSWORD);
+        return getString(HTTP_SERVER_TRUST_STORE_PASSWORD);
```
Comment (nicoloboschi) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java:3666]: should we handle retrocompatibility here? the risk is that the user upgrades and the webserver doesn't work anymore

at least, we should add highlight in the release notes

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Review by hangc0276 — COMMENTED** [review]
I think we should handle the  retro compatibility

**Comment by hangc0276** [comment]
@merlimat @eolivelli @dlg99 @zymap Please help take a look, thanks.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java (hunk 1 above)
Comment (nicoloboschi) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java:3666]: Please add fallback to the old property. Also a comment in the code pointing to this pull would be valuable

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by wenbingshen** [comment]
Maybe we can mark this config as deprecated and add a new config to restore what it actually does and doc it in the docs and release notes.

**Comment by shoothzj** [comment]
@wenbingshen any updates?

**Comment by shoothzj** [comment]
closed by #4301

**Pull request closed without merging**